Fix percent encoded URLs causing the program to crash

oliveroneill opened the pull request:
I specifically had an issue where a request of "/?test={encoded}" due to `{` being an invalid URL character. AWS Lambda Proxy Integration automatically decodes percent encodings and these needs to be re-encoded before creating the request. I've added a test for this specific case, it will crash without my change.

noppoMan commented:
Sorry for late reply🙇
I'd like to merge this PR! So, Could you resolve conflicts?
(Recently I updated HexavilleFramework to adopt swift-nio instead of Prorsum)

oliveroneill committed:
Fix percent encoded URLs causing the program to crash

The program will crash due to a URL with invalid characters
causing an error. AWS Lambda Proxy Integration decodes percent encodings
and these needs to be re-encoded before creating the request.

oliveroneill commented:
Done! Thanks @noppoMan
